src/tests/ftest/get_remote_files.sh

```diff
@@ -0,0 +1,374 @@
+    FILES_TO_PROCESS = ${FILES_TO_PROCESS}
+    CART_LOGTEST = ${CART_LOGTEST}
+    VERBOSE = ${VERBOSE}"
+}
+
+#######################################
+# Check if the files exist on this host.
+# Arguments:
+#   $1: local logs to process
+# Returns:
+#   0: Files exist or no matches were found
+#   1: Missing the log files argument
+#######################################
+check_files_input() {
+    if [ -n "${1}" ]; then
+        # shellcheck disable=SC2086
+        files=$(ls -d ${1} 2> /dev/null)
+        files_rc=$?
+        if [ -n "${files}" ] && [ "${files_rc}" -eq 0 ]; then
+            echo "Processing the following detected files:"
+            for file in ${files}
+            do
+                echo "  ${file}"
+            done
+        else
+            echo "No files matched ${1}. Done."
+            exit 0
+        fi
+    else
+        echo "Please specify -f option."
+        exit 1
+    fi
+}
+
+#######################################
+# Convert human readable size values to respective byte value.
+# Arguments:
+#   size: size value to convert to bytes. i.e. 5mb, 40gb, 6k, 10g
+# Returns:
+#   converted byte value.
+#######################################
+human_to_bytes() {
+    h_size=${1,,}
+
+    declare -A units=(
+        [b]=1
+        [kb]=$(python -c "print (2**10)")
+        [k]=$(python -c "print (2**10)")
+        [mb]=$(python -c "print (2**20)")
+        [m]=$(python -c "print (2**20)")
+        [gb]=$(python -c "print (2**30)")
+        [g]=$(python -c "print (2**30)")
+    )
+    if [[ "${h_size,,}" =~ ([0-9.]+)([kbmgt]+) ]]; then
+        val="${BASH_REMATCH[1]}"
+        unit="${BASH_REMATCH[2]}"
+        (( val = val*${units[${unit}]} ))
+        echo "${val}"
+    else
+        echo "Invalid value provided: ${1}"
+        return 1
+    fi
+}
+
+#######################################
+# List and tag files with a 'Y' or 'N' to classify
+# big files based on the threshold provided.
+# Arguments:
+#   logs_dir: path to local logs
+#   threshold: value to use as threshold to classify big files
+# Returns:
+#   stdout output listing the files prepended with a 'Y' or 'N' tag
+#######################################
+list_tag_files() {
+    # shellcheck disable=SC2016,SC1004,SC2086
+    awk_cmd='{ \
+    if (th <= $1){ \
+        print "  Y:",$1,$2} \
+    else { \
+        print "  N:",$1,$2}\
+    }'
+
+    # Convert to bytes and run
+    bts=$(human_to_bytes "${2}")
+    # shellcheck disable=SC2086
+    du -ab -d 1 ${1} | awk -v th="${bts}" "${awk_cmd}"
+}
+
+#######################################
+# Check if this host is a VM.
+# Arguments:
+#   $1: whether or not to skip the check
+# Returns:
+#   0: VM detected
+#   1: No VM detected or skipped
+#######################################
+check_hw() {
+    if [ "${1}" == "true" ]; then
+        echo "Skipping VM check for compression"
+        return 1
+    fi
+    dmesg | grep "Hypervisor detected" 2> /dev/null
+    return $?
```

Comment:
This is superfluous.  Functions return with their final command's exit status.

Reply:
In order to cleanup some messaging this function was reworked.  With the new code I don't believe this issue exists anymore.